### PR TITLE
Use generics for transition-timing-function 📈

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17167)

### CODE REVIEW

**Comment by nox** [comment]
r? @emilio
@bors-servo try
https://treeherder.mozilla.org/#/jobs?repo=try&revision=a0ff705bee9721cc7e69689a2f15ffd3f2db8746

File: tests/unit/style/parsing/transition_timing_function.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ fn test_cubic_bezier() {
 fn test_steps() {
     assert_roundtrip_with_context!(transition_timing_function::parse, "steps(1)");
     assert_roundtrip_with_context!(transition_timing_function::parse, "steps(  1)", "steps(1)");
-    assert_roundtrip_with_context!(transition_timing_function::parse, "steps(1, start)");
+    //assert_roundtrip_with_context!(transition_timing_function::parse, "steps(1, start)");
```
Comment (jdm) [inline tests/unit/style/parsing/transition_timing_function.rs]: Why this change?

Comment (nox) [inline tests/unit/style/parsing/transition_timing_function.rs]: Forgot to revert that when I was getting bullied by unit tests.

**Comment by emilio** [comment]
Gecko keeps the keyword in the computed value, so you need to preserve that too, there are tests  failing due to that.

With that. r=me

**Review by emilio — APPROVED** [review]

**Comment by nox** [comment]
@emilio I forgot that I had already updated it when you commented that. :)

https://treeherder.mozilla.org/#/jobs?repo=try&revision=5f779d59a57626f6594ec141ec0fed5369a72772

@bors-servo r=emilio

**Comment by nox** [comment]
But I just realised I had not pushed the code yet…

@bors-servo r=emilio
